File: back/src/main/java/org/example/skillandyou/document/Notification.java
package org.example.skillandyou.document;


import jakarta.persistence.Id;
import lombok.*;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;

@Document(collection = "notifications")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Notification {

    @Id
    private String id;

    private Long userId;
    private String type;
    private String message;


    private Long exchangeId;
    private String senderName;

    private Boolean read = false;
    private LocalDateTime createdAt = LocalDateTime.now();
}

File: front/src/types/Notification.ts
export interface Notification {
    id: string;
    userId: number;
    type: 'EXCHANGE_CREATED' | 'EXCHANGE_ACCEPTED' | 'EXCHANGE_COMPLETED';
    message: string;
    exchangeId: number;
    senderName: string;
    read: boolean;
    createdAt: string;
}